boilerplate/openshift/golang-osd-operator/standard.mk

```
@@ -332,3 +333,27 @@ opm-build-push: docker-push
 .PHONY: ensure-fips
 ensure-fips:
 	${CONVENTION_DIR}/configure-fips.sh
+
+.PHONY: migrate-tools
+migrate-tools:
+	# Installing controller-gen v0.8.0
+	go install sigs.k8s.io/controller-tools/cmd/controller-gen@${CONTROLLER_GEN_VERSION}
+	mv ${GOBIN}/controller-gen ${GOBIN}/controller-gen-${CONTROLLER_GEN_VERSION}
+
+	# Installing openapi-gen v0.23.0
+	go install k8s.io/code-generator/cmd/openapi-gen@${OPENAPI_GEN_VERSION}
+	mv ${GOBIN}/openapi-gen ${GOBIN}/controller-gen-${CONTROLLER_GEN_VERSION}
+
+.PHONY: migrate-files
+# You will need to export the forked/cloned operator repository directory as REPO to make this work.
+# Example: export REPO=~/Projects/My-Operator-Fork
+migrate-files:
+	rm -rf config
+	rsync -a $(REPO)/deploy .
+	rsync -a $(REPO)/config/config.go config/
+	rsync -a $(REPO)/config/templates/csv-template.yaml config/templates/
+	rsync -a $(REPO)/pkg/ ./pkg/ --exclude={'apis','controller'}
+	rsync -a $(REPO)/ . --exclude={'cmd','version'} --ignore-existing
+
+.PHONY: migrate
+migrate: migrate-files migrate-tools generate
```

- Suggest naming this rule more descriptively, e.g. `migrate-to-osdk1`
- Per above, suggest running the `generate` through `container-make`:
```suggestion
migrate: migrate-files migrate-tools
	./boilerplate/_lib/container-make generate
```

---

Getting this weird error when trying to use `container-make`, any idea what might be causing this?

```
# Copying files & folders
rm -rf config
rsync -a /home/Supreeth/Work/Forks/pagerduty-operator/deploy . --exclude=crds 
rsync -a /home/Supreeth/Work/Forks/pagerduty-operator/config/config.go config/
rsync -a /home/Supreeth/Work/Forks/pagerduty-operator/config/templates/ config/templates/
rsync -a /home/Supreeth/Work/Forks/pagerduty-operator/pkg . --exclude={'apis','controller'}
rsync -a /home/Supreeth/Work/Forks/pagerduty-operator/ . --exclude={'cmd','version','boilerplate','deploy','pkg'} --ignore-existing
# Running make generate
./boilerplate/_lib/container-make generate

==============================
Starting the container
==============================
Error: stat .docker/config.json: no such file or directory
==ERROR== Couldn't start detached container
```

---

Also... I might be doing it wrong, but even though I've exported `OLD_SDK_REPO_DIR` it fails to parse in Makefile. Do I have to initialize it somewhere?

```
PHONY: migrate-to-osdk1
migrate-to-osdk1:
	ifndef OLD_SDK_REPO_DIR
	$(error OLD_SDK_REPO_DIR is not set)
	endif
	# Copying files & folders
	rm -rf config
	rsync -a $(OLD_SDK_REPO_DIR)/deploy . --exclude=crds 
	rsync -a $(OLD_SDK_REPO_DIR)/config/config.go config/
	...
```

```
make migrate-to-osdk1                                                                                                git:(master|✚8…18 
boilerplate/openshift/golang-osd-operator/standard.mk:342: *** OLD_SDK_REPO_DIR is not set.  Stop.
```

---

Think of `ifndef`/`endif` as "precompiler directives". They're not interpreted as execution lines in a rule, but rather when the Makefile is parsed/loaded. I'm not actually sure _what_ `make` is making of your rule as written, but left-justifying those two lines fixes it:
```make
PHONY: migrate-to-osdk1
migrate-to-osdk1:
ifndef OLD_SDK_REPO_DIR
	$(error OLD_SDK_REPO_DIR is not set)
endif
	...
```
What this is actually doing is changing the _content_ of your rule at "compile-time" depending on whether the variable is set. If the variable is set, that `error` line is included (which of course means the rest of the rule isn't reached); if unset, it's excluded (so only the remainder of the rule is processed).

As for your `container-make` problem: do you get the same error when you run `container-make` manually from the shell?

---

Thanks for that detailed explanation, it works now. I grow a bit wiser every time I open a PR here.
As for the `container-make` problem, it runs all fine when run locally/manually from the shell.

On second thought, instead of automating generating manifests, I could ask the user to manually run `./boilerplate/_lib/container-make generate` in the SOP. 
This way the user can double-check if all files/folders have been copied over and the directory structure looks the way it is expected to (I've seen some operators having a different directory structure). Once sure, they can go ahead with manually running the `generate` command, this gives me a perception of having more control.

---

If you're going that route, why not ask them to do a full bp [bootstrap](https://github.com/openshift/boilerplate/#quick-start) rather than copying the `boilerplate` directory in. The onboarding process does a little bit more than just lay down files under that directory.

---

So with that change, the migration process would be something like:
1. Bootstrap boilerplate by subscribing to the golang-osd-operator convention by default.
2. Run `make migrate-to-osdk1` to copy over files.
3. Run `container-make generate` to generate manifests.

Does that look feasible?

---

Yah, something like that. Could swap steps 1 and 2.